lib/backend/pgbk/pgbk.go

```diff
@@ -480,9 +480,19 @@ func (b *Backend) DeleteRange(ctx context.Context, startKey []byte, endKey []byt
 func (b *Backend) KeepAlive(ctx context.Context, lease backend.Lease, expires time.Time) error {
 	revision := newRevision()
 	updated, err := pgcommon.Retry(ctx, b.log, func() (bool, error) {
+		// value = value || '' forces a value update; if the value is TOASTed
+		// and doesn't change, it won't appear in the change feed
+		//
+		// TODO(espadolini): figure out if it's better to switch to REPLICA
+		// IDENTITY FULL and fill unchanged columns from the identity data; it's
+		// unclear if the benefits of having a more standard solution outweigh
+		// the penalty of essentially doubling the WAL writes for no good
+		// reason, but it might also become useful in the future to have backend
+		// events include the previous state for the item (i.e. it might become
+		// a good reason)
 		tag, err := b.pool.Exec(ctx,
-			"UPDATE kv SET expires = $1, revision = $2 WHERE key = $3 AND (expires IS NULL OR expires > now())",
-			zeronull.Timestamptz(expires.UTC()), revision, lease.Key)
+			"UPDATE kv SET value = value || $1, expires = $2, revision = $3 WHERE key = $4 AND (expires IS NULL OR expires > now())",
```

Comment:
Scary. How sure we can be the planner doesn't get smarter and optimize this one away too?

Reply:
"Not very", although it would be kind of weird for the prepared statement to special case the empty string as an input parameter. I looked for a "black box" identity function that was guaranteed to not be inspectable by the query planner, but I couldn't find one.

After the work for #29966 (if we decide to go for `REPLICA IDENTITY FULL`) we will be able to drop this.